cpp/src/arrow/compute/exec/aggregate_node.cc

```diff
@@ -440,15 +623,33 @@ class GroupByNode : public ExecNode, public TracedNode {
       int key_field_id = key_field_ids[i];
       output_fields[base + i] = input_schema->field(key_field_id);
     }
+    base += keys.size();
+    for (size_t i = 0; i < segment_keys.size(); ++i) {
+      int segment_key_field_id = segment_key_field_ids[i];
+      output_fields[base + i] = input_schema->field(segment_key_field_id);
```

Comment:
If a column was used both as a key field and a segment field does this mean we would output that column twice?

Reply:
Yes, but [as noted above](https://github.com/apache/arrow/pull/34311#discussion_r1116877818), I think this usage should be disallowed.